dynamiq/nodes/agents/base.py

```diff
@@ -493,14 +512,56 @@ def _get_tool(self, action: str) -> Node:
             )
         return tool
 
-    def _run_tool(self, tool: Node, tool_input: str, config, **kwargs) -> Any:
+    def _apply_parameters(self, merged_input: dict, params: dict, source: str, debug_info: list = None):
+        """Apply parameters from the specified source to the merged input."""
+        for key, value in params.items():
+            if key in merged_input and isinstance(value, dict) and isinstance(merged_input[key], dict):
+                merged_nested = merged_input[key].copy()
+                merged_input[key] = self.deep_merge(value, merged_nested)
+                if debug_info is not None:
+                    debug_info.append(f"  - From {source}: Merged nested {key}")
```

Comment:
- `debug_info` could be None, so init it in such case properly
- is this proper debug message here?